src/rubrix/listeners/listener.py

```diff
@@ -158,7 +159,9 @@ def __listener_iteration_job__(self, *args, **kwargs):
             name=self.dataset, task=dataset.task, query=self.formatted_query, size=0
         )
 
-        ctx.search = Search(total=search_results.total)
+        ctx.search = Search(
+            total=search_results.total, params=copy.deepcopy(ctx.query_params)
```

Comment:
Shouldn't this be:
```suggestion
            total=search_results.total, query_params=copy.deepcopy(ctx.query_params)
```
?? Not sure why the tests pass...

Reply:
Me either. I'll check that. Thanks !

Reply:
Error handling related errors. Now, the listener will stop if any unhandled error occurs. Please, take a look at the changes @dcfidalgo